### PR TITLE
Benchmark/pagefaults

### PR DESCRIPTION
**Pull request by phoeinx**

### CODE REVIEW

File: scripts/mp-2023/analyze_memory_peaks.py [hunk 1]
```diff
@@ -34,10 +35,14 @@
...
     print(f"{key} : {data[key]['memory_peak'] / GB}GB")
 
-# plot data values
-#     plt.plot(data[key]['memory_sizes'])
-#     plt.title(f'{key}GB')
-#     plt.show()
+#plot data values
+    plt.plot([value / GB for value in data[key]['memory_sizes']])
+    plt.title(f'Memory Footprint over Time for TPC-H Benchmark SF{key} on Hyrise Master')
+    plt.xlabel('Time matched to queries being executed for 80 seconds each')
```
Comment (SanJSp) [inline scripts/mp-2023/analyze_memory_peaks.py:41]: Should this also be title cased?

File: scripts/mp-2023/plot_multicore_performance_over_cores.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 import json
 
 #configure directory with benchmark results
-benchmark_results_dir = "./scripts/results_variable_core_new_master"
+benchmark_results_dir = "./scripts/benchmark_hyrise_master_variable_core_23_01_04"
```
Comment (SanJSp) [inline scripts/mp-2023/plot_multicore_performance_over_cores.py:15]: could be made more general

File: scripts/mp-2023/plot_performance_over_available_memory.py [hunk 3]
```diff
@@ -43,18 +54,27 @@
...
-                data_df = data_df.append({"memory": float(memory), "type": type, "latency": int(latency_all)}, ignore_index=True)
+                data_df = data_df.append({"memory": memory, "type": type, "latency": int(latency_all), "pgfaults": execution_pgfaults}, ignore_index=True)
+
 
-print(data_df)
 benchmark_results = sns.lineplot(data=data_df, x="memory", y="latency", hue="type", marker='o', linestyle='--')
+benchmark_results_second = sns.lineplot(data=data_df, x="memory", y="pgfaults", color='red', marker='o', linestyle='dotted', ax=benchmark_results.twinx())
 
 benchmark_results.set(
     xlabel="Available RAM in GB", ylabel="Latency in ms/iter (Sum over all Queries)", title=f"MMAP-based Single-Threaded Hyrise\nSum of Average Latency over all Queries Depending on Memory Limitation."
```
Comment (SanJSp) [inline scripts/mp-2023/plot_performance_over_available_memory.py:73]: For title case in all titles/axis labels
```suggestion
    xlabel="Available RAM in GB", ylabel="Latency in ms/iter (Sum over all Queries)", title=f"MMAP-Based Single-Threaded Hyrise\nSum of Average Latency over all Queries Depending on Memory Limitation."
```

File: scripts/mp-2023/plot_performance_over_available_memory.py [hunk 4]
```diff
@@ -43,18 +54,27 @@
...
 )
 
+benchmark_results_second.set(ylabel="#Pagefaults during Execution")
```
Comment (SanJSp) [inline scripts/mp-2023/plot_performance_over_available_memory.py:76]:
```suggestion
benchmark_results_second.set(ylabel="#Pagefaults During Execution")
```

File: scripts/mp-2023/analyze_memory_peaks.py (hunk 1 above)
Comment (SanJSp) [inline scripts/mp-2023/analyze_memory_peaks.py:41]:
```suggestion
    plt.title(fMemory Footprint Over Time for TPC-H Benchmark SF{key} on Hyrise Master')
    plt.xlabel('Time Matched to Queries Being Executed for 80 Seconds Each')
```

**Review by SanJSp — APPROVED** [review]
Only minor title casing

File: scripts/mp-2023/plot_multicore_performance_over_cores.py (hunk 2 above)
Comment (phoeinx) [reply scripts/mp-2023/plot_multicore_performance_over_cores.py:15]: always needs to be configured either way

**Pull request merged**